homeassistant/components/glances/sensor.py

```diff
@@ -79,30 +125,45 @@ def should_poll(self):
 
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
-        async_dispatcher_connect(
+        self.unsub_update = async_dispatcher_connect(
             self.hass, DATA_UPDATED, self._schedule_immediate_update
         )
 
     @callback
     def _schedule_immediate_update(self):
         self.async_schedule_update_ha_state(True)
 
+    async def will_remove_from_hass(self):
+        """Unsubscribe from update dispatcher."""
+        if self.unsub_update:
+            self.unsub_update()
+
     async def async_update(self):
         """Get the latest data from REST API."""
         value = self.glances_data.api.data
 
         if value is not None:
```

Comment:
Guard clause. `if value is None: return`

Reply:
done